### PR TITLE
Initial code removal of dead frontend code

### PR DESCRIPTION
**Pull request by jbradl11**
Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

Removed features and code that no longer are used in Bonnie:
- Debug Measure feature
- Complexity Dashboard
- Patient Dashboard
- Patient Bank
- Logic views for QDM Logic Measures
- QDM Calculator
- CQL Playground
- ValueSets Builder
- Asses related to Ace Text Editor
- Assets related to Datatables

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1263
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced.
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 
- [x] Automated regression test(s) pass

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint: N/A

**Reviewer 1:**

Name: @losborne
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

File: app/assets/javascripts/router.js.coffee.erb [hunk 1]
```diff
@@ -25,16 +24,10 @@
...
     'measures/:hqmf_set_id':                           'renderMeasure'
-    'measures/:hqmf_set_id/patient_dashboard':         'renderPatientDashboard'
-    'measures/:hqmf_set_id/508_patient_dashboard':     'renderPatientDashboard'
     'measures/:measure_hqmf_set_id/patients/:id/edit': 'renderPatientBuilder'
     'measures/:measure_hqmf_set_id/patients/new':      'renderPatientBuilder'
-    'measures/:measure_hqmf_set_id/patient_bank':      'renderPatientBank'
     'admin/users':                                     'renderUsers'
-    'value_sets/edit':                                 'renderValueSetsBuilder'
     # This will catch any unmatched hashes and route them to showPageNotFound
     '*path':                                            'showPageNotFound'
```
Comment (lukoktonos) [inline app/assets/javascripts/router.js.coffee.erb:33]: In this file on line 44-45, is `@isPortfolio` and `Thorax.Views.Matrix` still used, or can they be removed?

File: vendor/assets/components/ace-builds/demo/kitchen-sink/docs/Jack.jack [hunk 2]
```diff
@@ -1,247 +0,0 @@
...
--- Zip three different collections together
-p("zipped", i-collect(i-zip(
-  range(10),
-  [1,2,3,4,5],
-  i-map({i|i*i}, range(10))
-)))
-
-vars names, person, i, doubles, lengths, cubeRange
-names = ["Thorin", "Dwalin", "Balin", "Bifur", "Bofur", "Bombur", "Oin",
-         "Gloin", "Ori", "Nori", "Dori", "Fili", "Kili", "Bilbo", "Gandalf"]
```
Comment (lukoktonos) [inline vendor/assets/components/ace-builds/demo/kitchen-sink/docs/Jack.jack:28]: I don't know if we want to remove Gandalf...seems risky.

File: app/assets/javascripts/calculator_selector.js.coffee [hunk 3]
```diff
@@ -4,16 +4,11 @@
   calculate: (population, patient) ->
     if population.collection.parent.has('cql')
```
Comment (lukoktonos) [inline app/assets/javascripts/calculator_selector.js.coffee:5]: In other files you removed the check for `has('cql')`--do you think it makes sense to remove them in this file as well?

**Review by lukoktonos — COMMENTED** [review]
It is fairly difficult to review so many file changes, but I didn't find any files that were removed that caused any issues.  Both front-end and back-end test failures match the base branch.  

This PR is just a first pass, but as part of this review I enumerated the templates and classes and checked to see if they were used.  Here are the results of that analysis:

These classes seem to still be uninstantiated (thus potentially unused):
Thorax.Models.User
Thorax.Views.ValueSetCodesPageableList
Thorax.Views.MeasureRowView
Thorax.Views.PopulationView
Thorax.Views.PopulationTitle
Thorax.Views.MeasurePercentageView
Thorax.Views.MeasureStatusView
Thorax.Views.MeasureFractionView
Thorax.Views.RecordSection
Thorax.Views.RecordHeader

These templates seem to still be unused:
cql/cql_result_view.hbs
cql/cql_results_view.hbs
dashboard/compexity_popover.hbs
patients/patients.hbs
patients/record.hbs

If any of those stand out as being clear candidates for removal, perhaps they can be added to the list of files removed in this PR.  If there is any doubt, they could be considered instead in https://jira.mitre.org/browse/BONNIE-1267

**Comment by jbradl11** [comment]
Hold off on merging this.  I think I'm going to modify the PR, so it doesn't remove the vendor files (for readability of the PR).  We can do that in another PR. Thoughts?

**Comment by c-monkey** [comment]
if we do that, make sure to squash-merge this one in so those aren't in the history

**Comment by lukoktonos** [comment]
Another option could be to just separate this PR into 2 (or more) commits, so that each commit can be reviewed separately.

File: app/assets/javascripts/router.js.coffee.erb (hunk 1 above)
Comment (jbradl11) [reply app/assets/javascripts/router.js.coffee.erb:33]: @isPortfolio and Matrix are still used by the Cypress folks.

File: app/assets/javascripts/calculator_selector.js.coffee (hunk 3 above)
Comment (jbradl11) [reply app/assets/javascripts/calculator_selector.js.coffee:5]: Good catch, but I think we should hold off on making all of theses changes. I only removed ones that I knew wouldn't have any unintended consequences.

**Review by lukoktonos — APPROVED** [review]
Looks good.  Any failing tests match the base branch's.

**Pull request merged**